### PR TITLE
Allow passing in custom pricing in config_list

### PR DESCRIPTION
**Pull request by yiranwu0**
## Why are these changes needed?

Our price list is static and may not be up-to-date. We need a new PR each time for an updated price or a new model. 
In this PR, we allow the users to pass a `price` field which specify the price for a model in config_list. This allows user to set the price temporarily if we don't update on time. 

This is also a preliminary effort to incorporate an open-source model API calling infrastructure: https://deepinfra.com/models. They support inference of several open-source models, while using the OpenAI objects. We may not keep track of all model pricing, but the users can pass in prices as need while having access to all kinds of models.

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by yiranwu0** [comment]
@microsoft-github-policy-service agree

File: autogen/oai/client.py [hunk 1]
```diff
@@ -291,7 +291,9 @@ def cost(self, response: Union[ChatCompletion, Completion]) -> float:
         model = response.model
         if model not in OAI_PRICE1K:
             # TODO: add logging to warn that the model is not found
```
Comment (davorrunje) [inline autogen/oai/client.py:293]: should we remove TODO?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
-   && rm -rf /var/lib/apt/lists/* quarto-1.5.23-linux-${arch}.deb
-ENV DEBIAN_FRONTEND=dialog
-
-# For docs
-RUN npm install --global yarn
-RUN pip install --upgrade pip
-RUN pip install pydoc-markdown
-RUN pip install pyyaml
-RUN pip install colored
+#-------------------------------------------------------------------------------------------------------------
```
Comment (davorrunje) [inline .devcontainer/Dockerfile:1]: why is this shown as changed?

Comment (yiranwu0) [reply .devcontainer/Dockerfile:1]: I run `pre-commit` and this is what's changed.

File: autogen/oai/client.py (hunk 1 above)
Comment (yiranwu0) [reply autogen/oai/client.py:293]: Yeah.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (davorrunje) [reply .devcontainer/Dockerfile:1]: This is unusual. What is your platform? Do you use a devcontainer?

Comment (davorrunje) [reply .devcontainer/Dockerfile:1]: Did you run
```
pre-commit run --all-files
```

If so, you can revert back files that should not be touched and not run pre-commit manually. Pre-commit will be run by git commit command on changed files.

Comment (davorrunje) [reply .devcontainer/Dockerfile:1]: I found the problem with the existing repo and created a PR to fix it (https://github.com/microsoft/autogen/pull/2915).

**Review by davorrunje — APPROVED** [review]

**Comment by sonichi** [comment]
@yiranwu0 could you check the test failure? https://github.com/microsoft/autogen/actions/runs/9483342004/job/26160337349?pr=2902

**Comment by Deykeepcomn4me** [comment]
These are hackers hacking me please contact fbi

**Comment by r4881t** [comment]
There is a bug with this code. It seems to count the actual tokens instead of 1k tokens

**Pull request merged**